compiler/rustc_trait_selection/src/traits/project.rs

```diff
@@ -2090,3 +2095,15 @@ impl<'cx, 'tcx> ProjectionCacheKeyExt<'cx, 'tcx> for ProjectionCacheKey<'tcx> {
         })
     }
 }
+
+fn get_associated_const_value<'tcx>(
+    selcx: &mut SelectionContext<'_, 'tcx>,
+    alias_ct: ty::Const<'tcx>,
+    param_env: ty::ParamEnv<'tcx>,
+) -> ty::Const<'tcx> {
+    // FIXME(mgca): We shouldn't be invoking ctfe here, instead const items should be aliases to type
+    // system consts that we can retrieve with some `query const_arg_of_alias` query. Evaluating the
+    // constant is "close enough" to getting the actual rhs of the const item for now even if it might
+    // lead to some cycles
+    super::evaluate_const(selcx.infcx, alias_ct, param_env)
```

Comment:
`feature(associated_const_equality)` *might* be a bit more cycle prone now due to evaluating here instead of normalizing to the impl definition and *then* evaluating? All the tests still pass though so I guess its fine- and both this and the previous way it worked can't stick around long term anyway.